notes/dns-client-proxy.md

```diff
@@ -0,0 +1,667 @@
+are specified then the proxy assumes that no name is known (though
+a PKIX certificate may include an address literal in the subjectAltName).
+If both a name and address are specified then the proxy will use the
+specified address and use the name for authentication.
+
+To simplify the encoding of the option, an option with addresses will
+have either IPv4 or IPv6 addresses. If the application wants to specify
+both IPv4 and IPv6 addresses for a certain authentication-domain-name
+then it has to include two options. 
+
+An application may want to specify a DNS resolver that is reachable 
+through an IPv6 link-local address.
+IPv6 link-local addresses are special in that they require a zone to be
+specified, either explicitly or implicitly. Typically for a link-local
+address that appears as a source or destination address,
+the zone is implicitly the zone of the link the packet travels on.
+For packets that travel between hosts, there is no goed way to explictly
+specify the zone of a link-local address because two different hosts
+do not agree on zone names. However, if the proxy is on the same host
+as the application, then the zone identifier for the link-local address
+can be specified in the Interface field. For this purpose an interface
+name can also be an interface index expressed as a decimal string.
+
+
+When present, Service Parameters specify how to connect. Otherwise it is
+up to the proxy to try various possibilities. For Service Parameters,
+the values of the ipv4hint and ipv6hint fields are ignored. Addresses
+can only be specified using the addresses field in the PROXY CONTROL Option.
+
+Associated with this option is a new error, BADPROXYPOLICY. When
+a proxy cannot meet the requirements in a PROXY CONTROL Option or the
+option is malformed, it returns this error.
+
+# PROXY SCOPE OPTION
+
+~~~
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   0: |                          OPTION-CODE                          |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   2: |                         OPTION-LENGTH                         |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+   4: |                          Scope                                |
+      +---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+---+
+~~~
+
+{newline="true"}
+OPTION-CODE
+
+: To be decided
+
+OPTION-LENGTH
+
+: Length of this option excluding the OPTION-CODE and OPTION-LENGTH fields
+
+Scope
+
+: Scope of the source address of a request. Scope can have the following
+values:
+
+     Value | Scope
+    -------|-------
+     0     | Undefined
+     1     | Host local
+     2     | Link local
+     3     | Site local
+     4     | Global
+
+The purpose of this option is to deal with proxies that forward
+DNS traffic without first removing any EDNS(0) options. The option requests
+the DNS proxy that processes the option to report the scope of the source
+address. 
```

Comment:
I see you removed the text "The application can specify that the connection between the stub resolver and the proxy should have, for example, at most host local scope.". Can clients no longer restrict the scope?

Reply:
Yes. I decided that application can check, but the proxy does not enforce. That simplifies things because the proxy just includes the option with the right value, no action needs to be taken by the proxy.